app/Http/Controllers/Admin/UserController.php

```diff
@@ -189,6 +208,39 @@ public function update(UserFormRequest $request, User $user)
         return redirect()->route('admin.users.view', $user->id);
     }
 
+    /**
+     * Change a user to OAuth2/Normal on the system.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Pterodactyl\Models\User $user
+     * @return \Illuminate\Http\RedirectResponse
+     *
+     * @throws \Exception
+     * @throws \Throwable
+     */
+    public function put(Request $request, User $user)
```

Comment:
It looks like this function shouldn't even be called put, or should be in it's own controller under a different route because it is tied specifically to updating OAuth credentials, not the user.